todo_app/app.py

```diff
@@ -1,31 +1,56 @@
-from flask import Flask, render_template, redirect, request
-from todo_app.data.session_items import get_items, add_item
+from flask import Flask, appcontext_popped, render_template, redirect, request
 from todo_app.flask_config import Config
+from dotenv import load_dotenv, find_dotenv
+from todo_app.data.trello_items import *
+from todo_app.data.class_definitions import ViewModel
+import os
 
-app = Flask(__name__)
-app.config.from_object(Config())
-#@app.route('/')
-#def index():
-    # Modify the index() function to get the list of items 
- #  items = get_items() 
-  # return render_template('index.html', ls = items)
-
-@app.route('/', methods = ['GET', 'POST'])
-def post_item():
-   items = get_items()
-   if request.method == 'POST':
-      title =  request.form.get('name')
-      items = add_item(title)
-      print(items)
-      render_template('index.html', ls = items)
-      return redirect(request.url)
-   return render_template('index.html', ls = items) 
-  # return 'hello'
-# Update the function for your new route to retrieve the item title from the form data using request.form.get('field_name') — 
-# where field_name 
-# is the name of the input field in your form.
-# Finally, add the item to the list, and redirect the user back to the index page, using the redirect function from Flask. 
-# Returning a redirect means you avoid repetition of code, 
-# plus the browser will show the correct URL and can refresh without resubmitting the form.
+file_path = find_dotenv('.env')
+load_dotenv(file_path, override=True)
 
+def create_app():
+
+   token = os.getenv('TRELLO_TOKEN')
+   key = os.getenv('TRELLO_KEY')
+   board_ID = os.getenv('BOARD_ID')
+   url = "https://api.trello.com/1/"
+   url_member = url + "members/carispang"
+   query_string = {"key" : key, "token" : token}
+
+   app = Flask(__name__)
+   app = Flask(__name__, template_folder='templates')
+   app.config.from_object(Config())
+
+   @app.route('/', methods = ['GET', 'POST'])
+   def createUI():
+      data_board_lists = get_board_lists(url, query_string, board_ID)
+      list_id = get_list('To Do', data_board_lists)
+      todo_cards = get_cards_on_list(url, list_id, query_string)
+      if request.method == 'POST':
+         if not request.form.get('complete_button'):
+            card_name = request.form.get('submit_button') 
+            create_card(url, list_id, card_name, key, token)
+            todo_cards.append({"name": card_name})
+         else:
+            data_board_cards = get_data_board_cards(url, query_string)
+            data_board_lists = get_board_lists(url, query_string)
+            done_ID = get_list('Done', data_board_lists)
+            complete_method(data_board_cards, url, done_ID, key, token)
+            todo_cards = get_cards_on_list(url, list_id, query_string)
+      item_view_model = ViewModel(todo_cards)      
+      return render_template('index.html', view_model = item_view_model)
+
+   @app.route('/Module_3')
+   def module_3():
```

Comment:
We don't really want a new route and template for the module. You were meant to update your existing index.html and route